### PR TITLE
add Menoh API

### PR DESCRIPTION
**Pull request by okdshin**

### CODE REVIEW

File: menoh/menoh_chainer_compiler.cpp [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+
+            /*
+            if (!chainer_compiler::g_skip_inference) {
+                graph->InferShapes();
+            }
+            */
+
+            constexpr bool kBackprop = false;
+            chainer_compiler::RunDefaultPasses(&graph, kBackprop);
+            chainer_compiler::runtime::XCProgramProto chxvm_prog;
```
Comment (shinh) [inline menoh/menoh_chainer_compiler.cpp]: Note I'm planning to rename this to `ChxVMProgramProto`: https://github.com/pfnet-research/chainer-compiler/pull/533

Comment (okdshin) [reply menoh/menoh_chainer_compiler.cpp]: OK. I will modify. Thank you for the info.

**Comment by shinh** [comment]
ChainerCI, test this please

File: CMakeLists.txt [hunk 2]
```diff
@@ -475,3 +475,9 @@ gen_elichika_test(node/Functions/Activation "${CHAINER_COMPILER_TEST_ALL}")
 gen_elichika_test(node/Links "${CHAINER_COMPILER_TEST_ALL}")
 gen_elichika_test(syntax "${CHAINER_COMPILER_TEST_ALL}")
 gen_elichika_test(chainercv_model/resnet "${CHAINER_COMPILER_TEST_ALL}")
+
+# Menoh
+option(CHAINER_COMPILER_ENABLE_MENOH "Build Menoh API" ON)
```
Comment (shinh) [inline CMakeLists.txt]: I think we don't need an option to disable this.

File: menoh/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,63 @@
+get_filename_component(CHAINER_COMPILER_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR} PATH)
+add_custom_command(
+  OUTPUT
+    "${CMAKE_CURRENT_BINARY_DIR}/json_args.inc"
+  COMMAND
+    python3 "${CMAKE_CURRENT_SOURCE_DIR}/../scripts/generate_flags_code.py" --mode menoh_chainer_compiler.json_args.inc
```
Comment (shinh) [inline menoh/CMakeLists.txt]: s/python3/PYTHON_EXECUTABLE/ after https://github.com/pfnet-research/chainer-compiler/pull/559

**Comment by shinh** [comment]
Thanks!! I'll do these changes by myself.

**Comment by shinh** [comment]
https://github.com/pfnet-research/chainer-compiler/pull/560

**Pull request merged**